sdk/appconfiguration/app-configuration/samples/react-sample/src/components/Page.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+  Copyright (c) Microsoft Corporation.
+  Licensed under the MIT license.
+*/
+
+import React from "react";
+import { useEffect, useState } from "react";
+import {
+  Navbar,
+  Nav,
+  NavDropdown,
+  Jumbotron,
+  Alert,
+  Row,
+  Col,
+  Container
+} from "react-bootstrap";
+import { getEnvironmentVariable } from "../utils";
+import {
+  AppConfigurationClient,
+  featureFlagPrefix,
+  isFeatureFlag,
+  isFeatureFlagClientFilter
+} from "@azure/app-configuration";
+
+let client: AppConfigurationClient;
+
+const featureFlag1 = {
+  id: "react-app-feature-1",
+  description: "",
+  enabled: false,
+  conditions: {
+    client_filters: []
+  }
+};
+const featureFlag2 = {
+  id: "react-app-feature-2",
+  description: "",
+  enabled: true,
+  conditions: {
+    client_filters: [
+      {
+        name: "Microsoft.TimeWindow",
+        parameters: {
+          Start: "Mon, 12 Apr 2021 07:00:00 GMT",
+          End: "Wed, 14 Apr 2021 07:00:00 GMT"
+        }
+      }
+    ]
+  }
+};
+
+export default function Page(): JSX.Element {
+  const feature1 = "react-app-feature-1";
+  const feature2 = "react-app-feature-2";
+  const [feature1ClassName, setFeature1ClassName] = useState<string>("feature-button-hidden");
+  const [feature2ClassName, setFeature2ClassName] = useState<string>("feature-button-hidden");
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const getFeatureFlags = async (keys: string[]): Promise<void> => {
+    const [setting1, setting2] = await Promise.all(
+      keys.map((key: string) =>
+        client.getConfigurationSetting({
+          key: featureFlagPrefix + key
+        })
+      )
+    );
+    if (isFeatureFlag(setting1)) {
+      console.log(`${setting1.key} is enabled : ${setting1.enabled}`, setting1);
+      setFeature1ClassName(setting1.enabled ? "feature-button-visible" : "feature-button-hidden");
+    }
+    if (isFeatureFlag(setting2)) {
+      console.log(`${setting2.key} is enabled : ${setting2.enabled}`, setting2);
+      const clientFilter = setting2.conditions.clientFilters?.[0];
+      if (isFeatureFlagClientFilter("timeWindow", clientFilter)) {
+        const now = Date.now();
+        const withinRange =
+          now - Date.parse(clientFilter.parameters.start) > 0 &&
+          Date.parse(clientFilter.parameters.end) - now > 0;
+        setFeature2ClassName(withinRange ? "feature-button-visible" : "feature-button-hidden");
+      }
+    }
+  };
+
+  useEffect(() => {
+    const connectionString = getEnvironmentVariable("REACT_APP_APPCONFIG_CONNECTION_STRING");
```

Comment:
Does this part execute on the server?

Reply:
REACT_APP_APPCONFIG_CONNECTION_STRING variable will be accessible at the client end. But I think that's for the sample app since the intention is to show how the feature flags are used?

@maorleger thoughts on how to secure this?

Reply:
Yeah, it's always a tradeoff between keeping the samples as simple as possible and demonstrating best practices with regards to securing things. Putting secrets in environment variables is unfortunately insecure because they get embedded / inlined. But setting up AAD client-side introduces a bunch of concepts and setup, distracting from what this sample demonstrates.

What I've done in the past is put a note in the README, near where I talk about the environment variables mentioning that it's normally not best practice but we'd like to avoid the complexity of setting up AAD. Something like this maybe?


```
> Remember: we are using a connection string to keep this sample simple; however, React will embed the connection 
string into your published bundle which is not suitable for production as it may leak secrets. For production client side 
applications, you may be interested in the [@azure/identity][identity] package which provides a set of credential 
implementations for both NodeJS and the browser.
```

What do you think?

Reply:
Sounds good to me.

Reply:
Do we have a sample of how to do AAD for client-side apps?

Reply:
Yes, we do.
https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/appconfiguration/app-configuration/README.md#authenticating-with-a-service-principal

Reply:
I think that particular method is really for the backend, since it relies on environment variables/endpoints that only exist on the service side.

Reply:
This might be an instance where you'd want to actually include the workflow for using the interactive browser credential. I don't think we ever want to encourage a connection string in the front-end.

Reply:
Okay, I'll switch to AAD.

Reply:
Switched to AAD

Reply:
Ping @richardpark-msft @pakrym